servers/cu/src/domain/event-transport.js

```diff
@@ -0,0 +1,101 @@
+import Libhoney from 'libhoney'
+import Database from 'better-sqlite3'
+import fs from 'fs'
+
+export class CompositeTransport {
+  constructor (transports) {
+    this.transports = transports // An array of transport instances
+  }
+
+  sendEvents (events, processId, nonce) {
+    this.transports.forEach((transport) => {
+      transport.sendEvents(events, processId, nonce)
+    })
+  }
+}
+
+export class ConsoleTransport {
+  sendEvents (events, processId, nonce) {
+    console.log(
+      `[ProcID: ${processId}; Nonce: ${nonce}]: Vacuumed events:`,
+      events
+    )
+  }
+}
+
+export class HoneycombTransport {
+  constructor (writeKey, dataset, apiHost = 'https://api.honeycomb.io', dbFilePath = './largest_nonces.db') {
+    this.apiHost = apiHost
+    this.writeKey = writeKey
+    this.honey = new Libhoney({
+      writeKey,
+      dataset,
+      apiHost
+    })
+    this.dbFilePath = dbFilePath
+    this.dbInitialized = false
+    this.db = undefined
+  }
+
+  ensureLargestNonces () {
+    if (!this.dbInitialized) {
+      console.log('[hc-transport]: Initializing SQLite database...')
+      if (!fs.existsSync(this.dbFilePath)) {
+        this.createDatabase()
+      } else {
+        this.db = new Database(this.dbFilePath)
+      }
+      this.dbInitialized = true
+    }
+  }
+
+  createDatabase () {
+    console.log('[hc-transport]: Creating, if necessary, SQLite database and table for largest nonces...')
+    this.db = new Database(this.dbFilePath)
+    this.db.exec(`
+      CREATE TABLE IF NOT EXISTS largest_nonces (
+      processId TEXT PRIMARY KEY,
+      nonce INTEGER
+      );
+    `)
+  }
+
+  getLargestNonce (processId) {
+    const stmt = this.db.prepare('SELECT nonce FROM largest_nonces WHERE processId = ?')
+    const row = stmt.get(processId)
+    return row ? row.nonce : 0
+  }
+
+  updateLargestNonce (processId, nonce) {
+    const stmt = this.db.prepare(`
```

Comment:
Same here as above re statement memoization.

Reply:
We still need to write to the db for durability but we also need to update the memo here.
9bdd5330

Reply:
0f2f6ac